util/jsonrpc-types/src/block_template.rs

```diff
@@ -6,22 +6,83 @@ use ckb_types::{packed, prelude::*, H256};
 use serde::{Deserialize, Serialize};
 use std::convert::From;
 
+/// A block template for miners.
+///
+/// Miners optional pick transactions and then assemble the final block.
 #[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Hash, Debug)]
 pub struct BlockTemplate {
+    /// Block version.
+    ///
+    /// Miners must use it unchanged in the assembled block.
     pub version: Version,
+    /// The compacted difficulty target for the new block.
+    ///
+    /// Miners must use it unchanged in the assembled block.
     pub compact_target: Uint32,
+    /// The timestamp for the new block.
+    ///
+    /// CKB node guarantees that this timestamp is larger than the median of previous 37 blocks.
+    ///
+    /// Miners can increase it to the current time. It is not recommended to decrease it, since it may vialate the median block timestamp consensus rule.
```

Comment:
```suggestion
    /// Miners can increase it to the current time. It is not recommended to decrease it, since it may violate the median block timestamp consensus rule.
```